dm/regexprrouter/router_test.go

```diff
@@ -0,0 +1,243 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package regexprrouter
+
+import (
+	"fmt"
+	"testing"
+
+	router "github.com/pingcap/tidb-tools/pkg/table-router"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCreateRouter(t *testing.T) {
+	_, err := NewRegExprRouter(true, []*TableRule{})
+	assert.Equal(t, nil, err)
+	_, err = NewRegExprRouter(false, []*TableRule{})
+	assert.Equal(t, nil, err)
+}
+
+func TestAddRule(t *testing.T) {
+	r, err := NewRegExprRouter(true, []*TableRule{})
+	assert.Equal(t, nil, err)
+	rules := []*TableRule{
+		{
+			SchemaPattern: "test1",
+			TargetSchema:  "dtest1",
+		},
+		{
+			SchemaPattern: "test2",
+			TablePattern:  "table2",
+			TargetSchema:  "dtest2",
+			TargetTable:   "dtable2",
+		},
+	}
+	for _, rule := range rules {
+		err = r.AddRule(rule)
+		assert.Equal(t, nil, err) // successfully insert
+	}
+	r, err = NewRegExprRouter(false, []*TableRule{})
+	assert.Equal(t, nil, err)
+	for _, rule := range rules {
+		err := r.AddRule(rule)
+		assert.Equal(t, nil, err) // successfully insert
+	}
+}
+
+func TestSchemaRoute(t *testing.T) {
```

Comment:
maybe add some case for ErrWorkerRouteTableDupMatch

Reply:
checked